mypy/dmypy_server.py

```diff
@@ -280,6 +280,9 @@ def initialize_fine_grained(self, sources: List[mypy.build.BuildSource]) -> Dict
             result = mypy.build.build(sources=sources,
                                       options=self.options,
                                       alt_lib_path=self.alt_lib_path)
+            # build will clear use_fine_grained_cache if it needs to give up
```

Comment:
Style nit: looks like it would be better to move this outside the try statement, since this can't generate a `CompileError` exception.